content/en/contribute/medic/product-development-process/_index.md

```diff
@@ -0,0 +1,14 @@
+---
+title: "Product Development Process"
+linkTitle: "Product Development Process"
+weight: 1
+description: >
+  Explanation of the Continuous Discovery process used for product development at Medic
+---
+
+The Medic Product Team operates in a way that is most commonly referred to as [Continuous Discovery](/contribute/medic/product-development-process/continuous-discovery-overview). That process helps us to focus on the impact of our work and keep teammates well-aligned with the people who use the CHT.
```

Comment:
For internal links throughout this PR it's better to use `{{% ref "..." $}}` rather than a regular link. This provides better checking for broken links at build time.

```suggestion
The Medic Product Team operates in a way that is most commonly referred to as [Continuous Discovery]({{% ref "continuous-discovery-overview" %}}). That process helps us to focus on the impact of our work and keep teammates well-aligned with the people who use the CHT.
```